neurom/io/datawrapper.py

```diff
@@ -72,11 +75,17 @@ def _section_end_points(data_block):
     # number of children per point
     n_children = defaultdict(int)
     for row in data_block:
-        n_children[int(row[COLS.P])] += 1
+            n_children[int(row[COLS.P])] += 1
 
     # end points have either no children or more than one
-    return set(i for i, row in enumerate(data_block)
-               if n_children[row[COLS.ID]] != 1)
+    endpoints = set(i for i, row in enumerate(data_block)
+                    if n_children[row[COLS.ID]] != 1)
+
+    #manually add the 'last' point of the soma
+    last_soma_idx = np.nonzero(data_block[:, COLS.TYPE] == POINT_TYPE.SOMA)[-1]
```

Comment:
The soma could actually be split into blocks, so the "last" soma point would only catch one of the end-points.